src/Datadog.Trace/Agent/MovingAverageKeepRateCalculator.cs

```diff
@@ -0,0 +1,175 @@
+﻿using System;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.Agent
+{
+    /// <summary>
+    /// Used to calculate the Trace Keep Rate, tracking the number of
+    /// traces kept and dropped that should have been sent to the agent.
+    /// Traces that are subsequently dropped by the agent due to sampling
+    /// will not count as dropped in this rate.
+    /// </summary>
+    internal class MovingAverageKeepRateCalculator : IKeepRateCalculator
+    {
+        private const int DefaultWindowSize = 10;
+        private static readonly TimeSpan DefaultBucketDuration = TimeSpan.FromSeconds(1);
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<MovingAverageKeepRateCalculator>();
+
+        private readonly int _windowSize;
+        private readonly TimeSpan _bucketDuration;
+        private readonly uint[] _dropped;
+        private readonly uint[] _created;
+
+        private readonly CancellationTokenSource _cts = new();
```

Comment:
I don't see this being used anywhere, can it be removed?

Reply:
Oops, yes, usage was replaced in https://github.com/DataDog/dd-trace-dotnet/pull/1306/commits/8302920160132346da769025b19ba66e3be9ec45. Removed in https://github.com/DataDog/dd-trace-dotnet/pull/1306/commits/7abb3b19a8c5637ef73a9775bffa7d5aed7b83c8